flow/include/flow/CompressionUtils.h

```diff
@@ -29,6 +29,7 @@ enum class CompressionFilter {
 #ifdef ZLIB_LIB_SUPPORTED
 	GZIP,
 #endif
+	ZSTD,
```

Comment:
Shouldn't this be wrapped in ZSTD_LIB_SUPPORTED?
Also I'm seeing a problem here, we could potentially have different enum values here depending on how fdb is compiled, which would be very bad if that changed.
We should probably always include them in the enum and have some other check to see if you are reading an unsupported compression filter type

Reply:
Yeah seems like the fallout. 

Valid concern, multiple ways to solve the issue:
1. Assign values to the enum variable, hence, making filter to enum value definite.
2. Include in enum regardless of the underlying library support. 


My inclination is to pursue #1, not having those values defined at the compile time is easy to catch unexpected behavior due to mis-configuration or programming logic errors, imho.

Reply:
Also the current mechanism allows catching the issue at the compile time which seems little better.

Reply:
However, `#ifdef` is used in too many files than what I would like, am working on refactoring code, such that CompressionUtils should be concerned about `filters` supported and given the filter decision is compile time, other user classes might always find out the set of supported filter. Shall update the code in a bit. 

Thanks

Reply:
Addressed the review comment as well did minor refactoring to limit `ZSTD_LIB_SUPPORTED` usage to CompressionUtils as much as possible.

Thanks

Reply:
I like this change a lot!